### PR TITLE
v053j

### PR DESCRIPTION
**Pull request by hussainmohd-a**
Changes related to v053j release

### CODE REVIEW

**Comment by ignoramous** [comment]
Can we make IPv4 the default?

**Comment by hussainmohd-a** [comment]
> 

Ya, we will make IPv4 as default for this version.

**Review by ignoramous — APPROVED** [review]

File: app/src/main/java/com/celzero/bravedns/service/PersistentState.kt [hunk 1]
```diff
@@ -197,7 +197,7 @@ class PersistentState(context: Context) : SimpleKrate(context), KoinComponent {
     var periodicallyCheckBlocklistUpdate by booleanPref("check_blocklist_update", false)
 
     // user-preferred Internet Protocol type, default IPv4
-    var internetProtocolType by intPref(INTERNET_PROTOCOL, InternetProtocol.IPv46.id)
+    var internetProtocolType by intPref(INTERNET_PROTOCOL, InternetProtocol.IPv4.id)
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/service/PersistentState.kt:200]: String needs to be changed too. Right now, its *Auto (default)*.

File: app/src/main/java/com/celzero/bravedns/adapter/ConnectionTrackerAdapter.kt [hunk 2]
```diff
@@ -142,7 +149,10 @@ class ConnectionTrackerAdapter(private val context: Context) :
             // known ports(reserved port and protocol identifiers).
             // https://github.com/celzero/rethink-app/issues/42 - #3 - transport + protocol.
             val resolvedPort = KnownPorts.resolvePort(port)
-            b.connLatencyTxt.text = if (resolvedPort != KnownPorts.PORT_VAL_UNKNOWN) {
+            // case: for UDP/443 label it as HTTP3 instead of HTTPS
+            b.connLatencyTxt.text = if (port == KnownPorts.HTTPS_PORT && proto == Protocol.UDP.protocolType) {
+                HTTP3
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/adapter/ConnectionTrackerAdapter.kt:154]: should it be moved to `strings.xml` or Constants?

File: app/src/main/java/com/celzero/bravedns/download/AppDownloadManager.kt [hunk 3]
```diff
@@ -143,7 +171,7 @@ class AppDownloadManager(private val context: Context,
         }
         val appVersionCode = persistentState.appVersion
         val url = "${Constants.ONDEVICE_BLOCKLIST_UPDATE_CHECK_URL}$timestamp&${Constants.ONDEVICE_BLOCKLIST_UPDATE_CHECK_PARAMETER_VCODE}$appVersionCode"
-        Log.d(LOG_TAG_DOWNLOAD, "Check for download 1, download type ${type.name} url: $url")
+        Log.d(LOG_TAG_DOWNLOAD, "Check for download, download type ${type.name} url: $url")
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/download/AppDownloadManager.kt:174]: remove?

File: app/src/main/java/com/celzero/bravedns/glide/RethinkGlideModule.kt [hunk 4]
```diff
@@ -39,11 +39,11 @@ class RethinkGlideModule : AppGlideModule() {
         val memoryCacheSizeBytes = 1024 * 1024 * 30 // 30 MB
         builder.setMemoryCache(LruResourceCache(memoryCacheSizeBytes.toLong()))
 
-        val diskCacheSizeBytes = 1024 * 1024 * 30 // 30 MB
+        val diskCacheSizeBytes = 1024 * 1024 * 350 // 350 MB
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/glide/RethinkGlideModule.kt:42]: Pls add a code comment

```
 // 350M disk cache is enough to store 35K favicons of 10KB size on average.
 // 35K is a good guesstimate of unique domains a device may hit over 30 days
 // Lowering disk-cache means an increase in data use (some reported, 600MB+)
```

File: app/src/main/java/com/celzero/bravedns/glide/RethinkGlideModule.kt [hunk 5]
```diff
@@ -39,11 +39,11 @@ class RethinkGlideModule : AppGlideModule() {
...
         builder.setDiskCache(InternalCacheDiskCacheFactory(context, diskCacheSizeBytes.toLong()))
 
         if (DEBUG) {
-            builder.setLogLevel(Log.DEBUG)
+            builder.setLogLevel(Log.ERROR)
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/glide/RethinkGlideModule.kt:46]: ?

File: app/src/main/java/com/celzero/bravedns/service/BraveVPNService.kt [hunk 6]
```diff
@@ -571,18 +573,17 @@ class BraveVPNService : VpnService(), ConnectionMonitor.NetworkListener, Blocker
...
+        val ts = underlyingNetworks?.lastUpdated
         // Added the INIT_TIME_MS check, encountered a bug during phone restart
         // isAccessibilityServiceRunning default value(false) is passed instead of
         // checking it from accessibility service for the first time.
-        if (activeNetworkHeartbeatTimestamp == INIT_TIME_MS || Math.abs(
-                now - activeNetworkHeartbeatTimestamp) > Constants.ACTIVE_NETWORK_CHECK_THRESHOLD_MS) {
-            Log.i(LOG_TAG_VPN,
-                  "Active network check, activeNetworkHeartbeatTimestamp: $activeNetworkHeartbeatTimestamp, is connection metered?: ${connectivityManager.isActiveNetworkMetered}," + "active network? ${connectivityManager.activeNetwork}, name: ${
+        if (ts == null || Math.abs(now - ts) > Constants.ACTIVE_NETWORK_CHECK_THRESHOLD_MS) {
+            Log.d(LOG_TAG_VPN,
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/service/BraveVPNService.kt:581]: ?

File: app/src/main/java/com/celzero/bravedns/service/ConnectionMonitor.kt [hunk 7]
```diff
@@ -164,7 +161,7 @@ class ConnectionMonitor(context: Context, networkListener: NetworkListener) :
 
     data class UnderlyingNetworks(val allNet: Set<Network>?, val ipv4Net: Set<Network>,
                                   val ipv6Net: Set<Network>, val useActive: Boolean,
-                                  var isActiveNetworkMetered: Boolean)
+                                                      var `isActiveNetworkMetered`: Boolean, var lastUpdated: Long)
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/service/ConnectionMonitor.kt:164]: What do backticks \` mean here for `isActiveNetworkMetered`?

File: app/src/main/java/com/celzero/bravedns/service/PersistentState.kt [hunk 8]
```diff
@@ -53,155 +54,179 @@ class PersistentState(context: Context) : SimpleKrate(context), KoinComponent {
...
+        "custom_downloader_last_generated_id").withDefault<Long>(0)
 
-    var isLocalBlocklistUpdateAvailable by booleanPref("local_blocklist_update_check", false)
+    // remove this var in 054
+    var isLocalBlocklistUpdateAvailable by booleanPref(
+        "local_blocklist_update_check").withDefault<Boolean>(false)
 
-    var isRemoteBlocklistUpdateAvailable by booleanPref("remote_blocklist_update_check", false)
+    // local timestamp for which the update is available
+    var updatableTimestampLocal by longPref("local_blocklist_update_ts").withDefault<Long>(
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/service/PersistentState.kt:199]: rename: `newestLocalBlocklistTimestamp`?

File: app/src/main/java/com/celzero/bravedns/service/PersistentState.kt [hunk 9]
```diff
@@ -53,155 +54,179 @@ class PersistentState(context: Context) : SimpleKrate(context), KoinComponent {
...
+        INIT_TIME_MS)
+
+    // remote timestamp for which the update is available
+    var updatableTimestampRemote by longPref("remote_blocklist_update_ts").withDefault<Long>(
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/service/PersistentState.kt:203]: rename: `newestRemoteBlocklistTimestamp` or something better?

File: app/src/main/java/com/celzero/bravedns/ui/AppConnectionBottomSheet.kt [hunk 10]
```diff
@@ -55,10 +52,27 @@ class AppConnectionBottomSheet(private val adapter: AppConnectionAdapter?, val u
...
+        const val IPADDRESS = "IPADDRESS"
+        const val IPRULESTATUS = "IPRULESTATUS"
+    }
+
     private fun isDarkThemeOn(): Boolean {
         return resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK == Configuration.UI_MODE_NIGHT_YES
     }
 
+    fun dismissListener(aca: AppConnectionAdapter?, pos: Int) {
+        adapter = aca
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/ui/AppConnectionBottomSheet.kt:72]: are we sure we are not leaking this?

File: app/src/main/java/com/celzero/bravedns/ui/AppConnectionBottomSheet.kt [hunk 11]
```diff
@@ -78,11 +92,20 @@ class AppConnectionBottomSheet(private val adapter: AppConnectionAdapter?, val u
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        uid = arguments?.getInt(UID) ?: -1
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/ui/AppConnectionBottomSheet.kt:95]: don't use magic ints (`-1` here). I believe there's already a `MISSING_UID` / `UNKNOWN_UID` or some such declared?

File: app/src/main/java/com/celzero/bravedns/ui/AppConnectionBottomSheet.kt [hunk 12]
```diff
@@ -78,11 +92,20 @@ class AppConnectionBottomSheet(private val adapter: AppConnectionAdapter?, val u
...
+        ipAddress = arguments?.getString(IPADDRESS) ?: ""
+        val status = arguments?.getInt(IPRULESTATUS) ?: IpRulesManager.IpRuleStatus.NONE.id
+        ipRuleStatus = IpRulesManager.IpRuleStatus.getStatus(status)
+
         initView()
         initializeClickListeners()
     }
 
     private fun initView() {
+        if (uid == -1) {
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/ui/AppConnectionBottomSheet.kt:105]: magic int `-1`.

File: app/src/main/java/com/celzero/bravedns/ui/AppConnectionBottomSheet.kt [hunk 13]
```diff
@@ -100,6 +123,14 @@ class AppConnectionBottomSheet(private val adapter: AppConnectionAdapter?, val u
         }
     }
 
+    override fun onResume() {
+        super.onResume()
+        if (uid == -1) {
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/ui/AppConnectionBottomSheet.kt:128]: magic int

File: app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt [hunk 14]
```diff
@@ -55,17 +55,23 @@ import com.celzero.bravedns.util.Utilities.Companion.getIcon
...
 
     private var _binding: BottomSheetConnTrackBinding? = null
 
     // This property is only valid between onCreateView and onDestroyView.
     private val b get() = _binding!!
 
+    private var ipDetails: ConnectionTracker? = null
+
+    companion object {
+        const val IPDETAILS = "IPDETAILS"
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt:72]: rename: `const val INSTANCE_STATE_IPDETAILS = "..."`

File: app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt [hunk 15]
```diff
@@ -92,8 +100,12 @@ class ConnTrackerBottomSheetFragment(private var ipDetails: ConnectionTracker) :
     }
 
     private fun initView() {
-        b.bsConnConnectionTypeHeading.text = ipDetails.ipAddress
-        b.bsConnConnectionFlag.text = ipDetails.flag
+        if (ipDetails == null) {
+            this.dismiss()
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt:104]: `Log.w("ip-details missing: initView called before onViewCreated?")`

File: app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt [hunk 16]
```diff
@@ -116,29 +128,35 @@ class ConnTrackerBottomSheetFragment(private var ipDetails: ConnectionTracker) :
 
     override fun onResume() {
         super.onResume()
+        if (ipDetails == null) {
+            this.dismiss()
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt:132]: `Log.w`

File: app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt [hunk 17]
```diff
@@ -116,29 +128,35 @@ class ConnTrackerBottomSheetFragment(private var ipDetails: ConnectionTracker) :
...
 
         b.bsConnDnsCacheText.visibility = View.VISIBLE
-        b.bsConnDnsCacheText.text = ipDetails.dnsQuery
+        b.bsConnDnsCacheText.text = ipDetails!!.dnsQuery
     }
 
     private fun updateConnDetailsChip() {
-        val protocol = Protocol.getProtocolName(ipDetails.protocol).name
-        val time = DateUtils.getRelativeTimeSpanString(ipDetails.timeStamp,
+        if (ipDetails == null) return
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt:151]: `Log.w`

File: app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt [hunk 18]
```diff
@@ -266,14 +286,14 @@ class ConnTrackerBottomSheetFragment(private var ipDetails: ConnectionTracker) :
...
                                         id: Long) {
-                val iv = view.findViewById<AppCompatImageView>(R.id.spinner_icon)
-                iv.visibility = View.VISIBLE
+                val iv = view?.findViewById<AppCompatImageView>(R.id.spinner_icon)
+                iv?.visibility = View.VISIBLE
                 val fid = IpRulesManager.IpRuleStatus.getStatus(position)
 
                 // no need to apply rule, prev selection and current selection are same
-                if (IpRulesManager.getStatus(ipDetails.uid, ipDetails.ipAddress) == fid) return
+                if (IpRulesManager.getStatus(ipDetails!!.uid, ipDetails!!.ipAddress) == fid) return
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt:296]: `ipDetails` could be a null pointer?

File: app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt [hunk 19]
```diff
@@ -356,28 +376,28 @@ class ConnTrackerBottomSheetFragment(private var ipDetails: ConnectionTracker) :
 
     private fun applyFirewallRule(firewallStatus: FirewallManager.FirewallStatus,
                                   connStatus: FirewallManager.ConnectionStatus) {
-        val appNames = FirewallManager.getAppNamesByUid(ipDetails.uid)
+        val appNames = FirewallManager.getAppNamesByUid(ipDetails!!.uid)
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt:379]: null ptr?

File: app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt [hunk 20]
```diff
@@ -356,28 +376,28 @@ class ConnTrackerBottomSheetFragment(private var ipDetails: ConnectionTracker) :
...
 
-        FirewallManager.updateFirewallStatus(ipDetails.uid, firewallStatus, connStatus)
+        FirewallManager.updateFirewallStatus(ipDetails!!.uid, firewallStatus, connStatus)
         updateFirewallRulesUi(firewallStatus, connStatus)
     }
 
     private fun applyIpRule(ipRuleStatus: IpRulesManager.IpRuleStatus) {
-        val protocol = Protocol.getProtocolName(ipDetails.protocol).name
-        val connRules = ConnectionRules(ipDetails.ipAddress, ipDetails.port, protocol)
+        val protocol = Protocol.getProtocolName(ipDetails!!.protocol).name
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt:391]: null ptr?

File: app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt [hunk 21]
```diff
@@ -356,28 +376,28 @@ class ConnTrackerBottomSheetFragment(private var ipDetails: ConnectionTracker) :
...
 
         Log.d(LOG_TAG_FIREWALL,
               "Apply ip rule for ${connRules.ipAddress}, ${FirewallRuleset.RULE2.name}")
-        IpRulesManager.updateRule(ipDetails.uid, connRules.ipAddress, ipRuleStatus)
+        IpRulesManager.updateRule(ipDetails!!.uid, connRules.ipAddress, ipRuleStatus)
     }
 
     private fun getAppName(): String {
-        val appNames = FirewallManager.getAppNamesByUid(ipDetails.uid)
+        val appNames = FirewallManager.getAppNamesByUid(ipDetails!!.uid)
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt:400]: null ptr?

File: app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt [hunk 22]
```diff
@@ -415,12 +435,13 @@ class ConnTrackerBottomSheetFragment(private var ipDetails: ConnectionTracker) :
             // b.root.invalidate()/ b.root.notify didn't help in this case.
             dialog.dismiss()
 
-            Log.d(LOG_TAG_FIREWALL, "Apply firewall rule for uid: ${ipDetails.uid}, ${status.name}")
-            FirewallManager.updateFirewallStatus(ipDetails.uid, status, connStatus)
+            Log.d(LOG_TAG_FIREWALL,
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt:438]: ?

File: app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt [hunk 23]
```diff
@@ -415,12 +435,13 @@ class ConnTrackerBottomSheetFragment(private var ipDetails: ConnectionTracker) :
...
+                  "Apply firewall rule for uid: ${ipDetails?.uid}, ${status.name}")
+            FirewallManager.updateFirewallStatus(ipDetails!!.uid, status, connStatus)
             updateFirewallRulesUi(status, connStatus)
         }.setNeutralButton(
             this.getString(R.string.ctbs_dialog_negative_btn)) { _: DialogInterface, _: Int ->
-            updateFirewallRulesUi(prevStatus, FirewallManager.connectionStatus(ipDetails.uid))
+            updateFirewallRulesUi(prevStatus, FirewallManager.connectionStatus(ipDetails!!.uid))
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt:444]: null ptr?

File: app/src/main/java/com/celzero/bravedns/ui/CustomIpActivity.kt [hunk 24]
```diff
@@ -15,44 +15,46 @@
...
 
     override fun onCreate(savedInstanceState: Bundle?) {
+        setTheme(Themes.getCurrentTheme(isDarkThemeOn(), persistentState.theme))
         super.onCreate(savedInstanceState)
-        setCancelable(true)
-        requestWindowFeature(Window.FEATURE_NO_TITLE)
-        b = DialogCustomIpBinding.inflate(layoutInflater)
-        setContentView(b.root)
-
-        window?.setLayout(WindowManager.LayoutParams.MATCH_PARENT,
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/ui/CustomIpActivity.kt:54]: curious: why remove?

File: app/src/main/java/com/celzero/bravedns/ui/DnsBlocklistBottomSheetFragment.kt [hunk 25]
```diff
@@ -61,22 +60,27 @@ import com.google.android.material.button.MaterialButton
...
 
+    private var transaction: DnsLog? = null
+
     private val persistentState by inject<PersistentState>()
 
     override fun getTheme(): Int = Themes.getBottomsheetCurrentTheme(isDarkThemeOn(),
                                                                      persistentState.theme)
 
+    companion object {
+        const val DNSLOG = "DNSLOGS"
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/ui/DnsBlocklistBottomSheetFragment.kt:81]: rename: `const val INSTANCE_STATE_DNSLOGS = "..."`

File: app/src/main/java/com/celzero/bravedns/ui/DnsBlocklistBottomSheetFragment.kt [hunk 26]
```diff
@@ -108,22 +121,26 @@ class DnsBlocklistBottomSheetFragment(private var contextVal: Context,
...
         }
 
         b.dnsRecordTypeChip.visibility = View.VISIBLE
-        b.dnsRecordTypeChip.text = getString(R.string.dns_btm_record_type, transaction.typeName)
+        b.dnsRecordTypeChip.text = getString(R.string.dns_btm_record_type, transaction!!.typeName)
     }
 
     private fun handleCustomDomainUi() {
-        when (DomainRulesManager.matchesDomain(transaction.queryStr)) {
+        if (transaction == null) return
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/ui/DnsBlocklistBottomSheetFragment.kt:141]: `Log.w`?

same for all

File: app/src/main/java/com/celzero/bravedns/ui/HomeScreenFragment.kt [hunk 27]
```diff
@@ -276,16 +277,13 @@ class HomeScreenFragment : Fragment(R.layout.fragment_home_screen) {
...
-            syncDnsStatus()
-            delay(TimeUnit.SECONDS.toMillis(2), lifecycleScope) {
-                b.fhsProxyChip.visibility = View.GONE
-                b.fhsProxyChip.isEnabled = true
-                showToastUiCentered(requireContext(),
-                                    getString(R.string.hsf_proxy_chip_removed_toast),
-                                    Toast.LENGTH_SHORT)
+            removeProxy()
+            io {
+                if(appConfig.isOrbotDns()) {
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/ui/HomeScreenFragment.kt:282]: `if<space>`

pls run the code formatter before check-in

File: app/src/main/java/com/celzero/bravedns/ui/SettingsFragment.kt [hunk 28]
```diff
@@ -847,7 +866,7 @@ class SettingsFragment : Fragment(R.layout.fragment_settings_screen) {
             }
         }
         io {
-            var proxyName = name
+            var proxyName = Constants.SOCKS
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/ui/SettingsFragment.kt:869]: is it a mistake to set `proxyName` always to `Constants.SOCKS`?

File: app/src/main/java/com/celzero/bravedns/ui/UniversalFirewallFragment.kt [hunk 29]
```diff
@@ -157,7 +157,8 @@ class UniversalFirewallFragment : Fragment(R.layout.universal_fragement_containe
         }
 
         includeView.firewallAppsShowTxt.setOnClickListener {
-            openCustomIpDialog()
+            enableAfterDelay(TimeUnit.SECONDS.toMillis(2L), includeView.firewallAppsShowTxt)
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/ui/UniversalFirewallFragment.kt:160]: isn't there a constant declared for this `SHORT_DELAY` or some such??

File: app/src/main/java/com/celzero/bravedns/ui/UniversalFirewallFragment.kt [hunk 30]
```diff
@@ -204,14 +205,10 @@ class UniversalFirewallFragment : Fragment(R.layout.universal_fragement_containe
...
-        val themeId = Themes.getCurrentTheme(isDarkThemeOn(), persistentState.theme)
-        val customDialog = CustomIpDialog(requireActivity(), customIpViewModel, themeId)
-        customDialog.show()
-    }
-
-    private fun isDarkThemeOn(): Boolean {
-        return resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK == Configuration.UI_MODE_NIGHT_YES
+    private fun openCustomIpScreen() {
+        val intent = Intent(requireContext(), CustomIpActivity::class.java)
+        intent.flags = Intent.FLAG_ACTIVITY_RESET_TASK_IF_NEEDED
```
Comment (ignoramous) [inline app/src/main/java/com/celzero/bravedns/ui/UniversalFirewallFragment.kt:210]: why is this flag used? Pls add a comment.

File: app/src/main/java/com/celzero/bravedns/adapter/AppConnectionAdapter.kt [hunk 31]
```diff
@@ -28,12 +29,17 @@ import com.celzero.bravedns.databinding.ListItemAppConnDetailsBinding
...
+class AppConnectionAdapter(val context: Context, connLists: List<AppConnections>,
                            val uid: Int) :
         RecyclerView.Adapter<AppConnectionAdapter.ConnectionDetailsViewHolder>(),
         AppConnectionBottomSheet.OnBottomSheetDialogFragmentDismiss {
 
     private lateinit var adapter: AppConnectionAdapter
+    private var ips: List<AppConnections>
+
+    init {
+        ips = connLists
```
[the remark this answers is not in the document]
Comment (hussainmohd-a) [reply app/src/main/java/com/celzero/bravedns/adapter/AppConnectionAdapter.kt:41]: The adapter requires different list to be assigned when filter is applied, so the `ips` are added in the adapter class

File: app/src/main/java/com/celzero/bravedns/adapter/ConnectionTrackerAdapter.kt [hunk 32]
```diff
@@ -97,7 +101,10 @@ class ConnectionTrackerAdapter(private val context: Context) :
                 return
             }
 
-            val bottomSheetFragment = ConnTrackerBottomSheetFragment(ct)
+            val bottomSheetFragment = ConnTrackerBottomSheetFragment()
+            val bundle = Bundle()
+            bundle.putString(ConnTrackerBottomSheetFragment.IPDETAILS, Gson().toJson(ct))
```
[the remark this answers is not in the document]
Comment (hussainmohd-a) [reply app/src/main/java/com/celzero/bravedns/adapter/ConnectionTrackerAdapter.kt:106]: Added comment.

File: app/src/main/java/com/celzero/bravedns/adapter/ConnectionTrackerAdapter.kt (hunk 2 above)
Comment (hussainmohd-a) [reply app/src/main/java/com/celzero/bravedns/adapter/ConnectionTrackerAdapter.kt:154]: Moved to strings.xml

File: app/src/main/java/com/celzero/bravedns/customdownloader/LocalBlocklistDownloader.kt [hunk 33]
```diff
@@ -94,6 +95,8 @@ class LocalBlocklistDownloader(val context: Context, workerParams: WorkerParamet
     private fun updatePersistenceOnCopySuccess(timestamp: Long) {
         persistentState.localBlocklistTimestamp = timestamp
         persistentState.blocklistEnabled = true
+        // reset updatable time stamp
+        persistentState.updatableTimestampLocal = INIT_TIME_MS
```
[the remark this answers is not in the document]
Comment (hussainmohd-a) [reply app/src/main/java/com/celzero/bravedns/customdownloader/LocalBlocklistDownloader.kt:99]: Yes, the boolean value for both remote and local will be removed in next version from PersistenceState.

File: app/src/main/java/com/celzero/bravedns/download/AppDownloadManager.kt (hunk 3 above)
Comment (hussainmohd-a) [reply app/src/main/java/com/celzero/bravedns/download/AppDownloadManager.kt:174]: Done

File: app/src/main/java/com/celzero/bravedns/glide/RethinkGlideModule.kt (hunk 4 above)
Comment (hussainmohd-a) [reply app/src/main/java/com/celzero/bravedns/glide/RethinkGlideModule.kt:42]: Done

File: app/src/main/java/com/celzero/bravedns/glide/RethinkGlideModule.kt (hunk 5 above)
Comment (hussainmohd-a) [reply app/src/main/java/com/celzero/bravedns/glide/RethinkGlideModule.kt:46]: Modified.

File: app/src/main/java/com/celzero/bravedns/service/ConnectionMonitor.kt (hunk 7 above)
Comment (hussainmohd-a) [reply app/src/main/java/com/celzero/bravedns/service/ConnectionMonitor.kt:164]: `isActiveNetworkMetered` is keyword used in ConnectivityManager, backticks will make sure it's local variable.

File: app/src/main/java/com/celzero/bravedns/service/PersistentState.kt (hunk 8 above)
Comment (hussainmohd-a) [reply app/src/main/java/com/celzero/bravedns/service/PersistentState.kt:199]: Renamed.

File: app/src/main/java/com/celzero/bravedns/service/PersistentState.kt (hunk 9 above)
Comment (hussainmohd-a) [reply app/src/main/java/com/celzero/bravedns/service/PersistentState.kt:203]: Renamed

File: app/src/main/java/com/celzero/bravedns/ui/AppConnectionBottomSheet.kt (hunk 11 above)
Comment (hussainmohd-a) [reply app/src/main/java/com/celzero/bravedns/ui/AppConnectionBottomSheet.kt:95]: Modified to use proper variable name.

File: app/src/main/java/com/celzero/bravedns/ui/AppConnectionBottomSheet.kt (hunk 12 above)
Comment (hussainmohd-a) [reply app/src/main/java/com/celzero/bravedns/ui/AppConnectionBottomSheet.kt:105]: Modified

File: app/src/main/java/com/celzero/bravedns/ui/AppConnectionBottomSheet.kt (hunk 13 above)
Comment (hussainmohd-a) [reply app/src/main/java/com/celzero/bravedns/ui/AppConnectionBottomSheet.kt:128]: Modified.

File: app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt (hunk 14 above)
Comment (hussainmohd-a) [reply app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt:72]: Renamed.

File: app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt (hunk 15 above)
Comment (hussainmohd-a) [reply app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt:104]: Done

File: app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt (hunk 16 above)
Comment (hussainmohd-a) [reply app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt:132]: added logs.

File: app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt (hunk 17 above)
Comment (hussainmohd-a) [reply app/src/main/java/com/celzero/bravedns/ui/ConnTrackerBottomSheetFragment.kt:151]: Done

File: app/src/main/java/com/celzero/bravedns/ui/CustomIpActivity.kt (hunk 24 above)
Comment (hussainmohd-a) [reply app/src/main/java/com/celzero/bravedns/ui/CustomIpActivity.kt:54]: Its moved from Dialog to Activity, Activity do not require window properties.

File: app/src/main/java/com/celzero/bravedns/ui/DnsBlocklistBottomSheetFragment.kt (hunk 25 above)
Comment (hussainmohd-a) [reply app/src/main/java/com/celzero/bravedns/ui/DnsBlocklistBottomSheetFragment.kt:81]: Renamed

File: app/src/main/java/com/celzero/bravedns/ui/HomeScreenFragment.kt (hunk 27 above)
Comment (hussainmohd-a) [reply app/src/main/java/com/celzero/bravedns/ui/HomeScreenFragment.kt:282]: Done

File: app/src/main/java/com/celzero/bravedns/ui/RethinkListFragment.kt [hunk 34]
```diff
@@ -180,6 +306,132 @@ class RethinkListFragment : Fragment(R.layout.fragment_rethink_list) {
         dialog.show()
     }
 
+    private fun initObservers() {
+        appDownloadManager.remoteDownloadStatus.observe(viewLifecycleOwner) {
+            Log.d(LoggerConstants.LOG_TAG_DOWNLOAD, "Remote blocklist download status id: $it")
```
[the remark this answers is not in the document]
Comment (hussainmohd-a) [reply app/src/main/java/com/celzero/bravedns/ui/RethinkListFragment.kt:311]: Some of the statements should be modified as Log.i. Modified those and added guard for the DEBUG logs.

File: app/src/main/java/com/celzero/bravedns/ui/SettingsFragment.kt (hunk 28 above)
Comment (hussainmohd-a) [reply app/src/main/java/com/celzero/bravedns/ui/SettingsFragment.kt:869]: Renamed the method name to `insertSocks5ProxyEndpointDB`. That method is only responsible for adding entry for SOCKS5 proxy.

File: app/src/main/java/com/celzero/bravedns/ui/UniversalFirewallFragment.kt (hunk 29 above)
Comment (hussainmohd-a) [reply app/src/main/java/com/celzero/bravedns/ui/UniversalFirewallFragment.kt:160]: No, we don't have that. Will introduce something like that. There are other places where we are using this. Will replace those as well after adding the constants.

File: app/src/main/java/com/celzero/bravedns/ui/UniversalFirewallFragment.kt (hunk 30 above)
Comment (hussainmohd-a) [reply app/src/main/java/com/celzero/bravedns/ui/UniversalFirewallFragment.kt:210]: Added comment.

**Pull request merged**